typescript/src/services/deposits/deposits-service.ts

```diff
@@ -12,12 +12,139 @@ import {
   BitcoinHashUtils,
   BitcoinLocktimeUtils,
   BitcoinScriptUtils,
+  BitcoinTxHash,
 } from "../../lib/bitcoin"
 import { Hex } from "../../lib/utils"
 import { Deposit } from "./deposit"
 import * as crypto from "crypto"
 import { CrossChainDepositor } from "./cross-chain"
 
+/**
+ * Result of initiating a gasless deposit where the relayer backend pays all
+ * gas fees.
+ *
+ * This structure contains both the Deposit object for Bitcoin operations and
+ * serializable data that can be stored (e.g., in localStorage) for later use
+ * in building the relay payload.
+ *
+ * @see {GaslessRevealPayload} for the payload structure needed after funding
+ */
+export interface GaslessDepositResult {
+  /**
+   * Deposit object for Bitcoin address generation and funding detection.
+   * Use `deposit.getBitcoinAddress()` to get the deposit address.
+   * Use `deposit.detectFunding()` to monitor for Bitcoin transactions.
+   */
+  deposit: Deposit
+
+  /**
+   * Deposit receipt containing all deposit parameters.
+   * This is serializable and can be stored for later payload construction.
+   */
+  receipt: DepositReceipt
+
+  /**
+   * Target chain name for the deposit.
+   * Can be "L1" or any L2 chain name (e.g., "Arbitrum", "Base", "Optimism").
+   */
+  destinationChainName: string
```

Comment:
Consider `type GaslessDestination = 'L1' | DestinationChainName`

Reply:
Good catch - implemented in #965336f2